### PR TITLE
AI-UserAPI Tech Design and Tech Spikes

### LINKED ISSUE #80: AI UserAPI Tech Design
## What is it
Write a technical design document using AI assistance for the Express UserAPI feature

# Context
The technical design will include the following sections:
- Goal
- Known Resources
- High Level Design
- Shapes
- Questions
- Test cases

# Acceptance Criteria
- Written in collaboration with ChatGPT
- Complete documentation of interaction with AI
[Requirements Document](https://docs.google.com/document/d/1WO-k2S2oAzRPPuPZk-0PPt2M-kNnWnLE-3TZUioUlQQ/edit?usp=sharing)

### PR DESCRIPTION
**Pull request by reimannsarah**
**What this is**
Complete Tech Design for overall project and Tech Spikes for Express.js and FileSystem.

**Context**
Tech Design and Spikes written with GitHub CoPilot. Include sections on known resources, shapes, approach, questions, and limitations. 

**Ticket**
[AI UserAPI Tech Design](https://github.com/orgs/olioapps/projects/11?pane=issue&itemId=37153935)

**Steps to Verify**
Navigate to ai-userAPI-tech-design branch and review:
- `README`
- `tech_design.md`
- `Express_tech_spike.md`
- `FileSystem_tech_spike.md`

**Media**
Screenshot 2023-08-29 at 2 52 19 PM

### CODE REVIEW

File: src/tech_design.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  * [Tech Spike: Using Express.js to Create a RESTful API](./tech_spike_express.md)
+  * [Express.js](https://expressjs.com/)
+  * [Express.js RESTful API](https://expressjs.com/en/starter/examples.html)
+  * [Express.js Static Files](https://expressjs.com/en/starter/static-files.html)
+  * Express.js is a Node.js module that allows for creating web applications. It is not a core module and must be installed via npm.
+* FileSystem
+  * [Tech Spike: FileSystem for Persistent Data Storage](./tech_spike_fs.md)
+  * [FileSystem](https://nodejs.org/api/fs.html)
+  * [FileSystem Read/Write](https://nodejs.org/api/fs.html#fs_fs_readfile_path_options_callback)
+  * FileSystem is a Node.js module that allows for reading and writing to the file system. It is a core module and does not need to be installed via npm.
```
Comment (DuHerb) [inline src/tech_design.md:34]: make this summary point the first bullet or inline with the 'FileSystem' header

File: src/tech_design.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+## Goal
+The goal of this project is to create a RESTful Express API that allows the storage and retrieval of user information. The user information will include fields such as name, age, favorite color, and an ID for each user.
+
+## Known Resources
+* Express.js
+  * [Tech Spike: Using Express.js to Create a RESTful API](./tech_spike_express.md)
+  * [Express.js](https://expressjs.com/)
+  * [Express.js RESTful API](https://expressjs.com/en/starter/examples.html)
+  * [Express.js Static Files](https://expressjs.com/en/starter/static-files.html)
+  * Express.js is a Node.js module that allows for creating web applications. It is not a core module and must be installed via npm.
```
Comment (DuHerb) [inline src/tech_design.md:29]: make this summary point the first bullet or inline with the 'Express.js' header

File: src/tech_design.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+* Retrieve user information for a specific user by ID.
+* Response: JSON object representing the user.
+
+`searchUsersByName` : GET /user
+* Retrieve user information for a specific name
+* Response: JSON object representing the user
+
+`searchUserByFavColor` : GET /user
+* Retriever user information for users with a specific favColor
+* Response: JSON object representing the User
```
Comment (DuHerb) [inline src/tech_design.md:66]: these two requests look like they'll include a body. What does that dataShape look like?

this helps define the 'contract' (or interface, or api ---- you'll hear lots of terms for this) between the client (the requestor) and the service (the provider).  

This way, if i gave this design to a front end dev, they could write a request in their UI without having to wait the service to be completed.

File: src/tech_design.md (hunk 1 above)
Comment (emmalgs) [reply src/tech_design.md:34]: fixed

File: src/tech_design.md (hunk 2 above)
Comment (emmalgs) [reply src/tech_design.md:29]: fixed

File: src/tech_design.md (hunk 3 above)
Comment (emmalgs) [reply src/tech_design.md:66]: fixed

**Pull request merged**